pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/stream/WritableStreamingData.java

```diff
@@ -100,11 +101,13 @@ public boolean hasRemaining() {
      * @return the actual number of bytes skipped.
      */
     @Override
-    public long skip(long count) {
+    public long skip(final long count) {
         try {
-            // We can only skip UP TO count, but if there are fewer bytes remaining, then we can only skip that many.
+            // We can only skip UP TO count.
             // And if the maximum bytes we can end up skipping is not positive, then we can't skip any bytes.
-            count = Math.min(count, remaining());
+            if (count > remaining()) {
+                throw new BufferUnderflowException();
```

Comment:
s/BufferUnderflowException/BufferOverflowException/ , since this happens on writing rather than on reading.

Reply:
Good catch. Fixed.